R/linting.R

```diff
@@ -0,0 +1,48 @@
+#' Use code linting
+#'
+#' `use_lintr()` Adds support for code linting using the `lintr` package in one of three ways:
+#' local interactive linting only, non-failing (post-build) Travis CI linting, or failing local and Travis CI `testthat` unit test.
+#'
+#' @param type Options are \code{local}, \code{travis} or \code{test}.
+#' @export
+use_lintr <- function(type = c("local", "travis", "test")){
+  type <- match.arg(type)
+
+  if(type == "test"){
+    dir.create("inst", showWarnings = FALSE)
```

Comment:
This should probably use the internal `create_directory()` or exported `use_directory()` for consistency with the rest of usethis.